packages/server/src/utils.ts

```diff
@@ -12,25 +12,30 @@ export function isAsyncIterable(body: any): body is AsyncIterable<any> {
 }
 
 export interface NodeRequest {
-  protocol?: string;
-  hostname?: string;
+  protocol?: string | undefined;
+  hostname?: string | undefined;
   body?: any;
-  url?: string;
-  originalUrl?: string;
-  method?: string;
+  url?: string | undefined;
+  originalUrl?: string | undefined;
+  method: string;
   headers?: any;
-  req?: IncomingMessage | Http2ServerRequest;
-  raw?: IncomingMessage | Http2ServerRequest;
-  socket?: Socket;
+  req?: IncomingMessage | Http2ServerRequest | undefined;
+  raw?: IncomingMessage | Http2ServerRequest | undefined;
+  socket?: Socket | undefined;
   query?: any;
 }
 
 export type NodeResponse = ServerResponse | Http2ServerResponse;
 
-function getPort(nodeRequest: NodeRequest) {
+function getPort(nodeRequest: NodeRequest | IncomingMessage | Http2ServerRequest | undefined) {
+  if (!nodeRequest) {
```

Comment:
Can we avoid functional changes, please :)

Reply:
we can't, const rawRequest: NodeRequest | Http2ServerRequest

Reply:
Sorry, I don't follow. It is `NodeRequest` already. In order to avoid type checking with `prop in obj`, we use `NodeRequest`. Adding more types affects the JS part. Why can't we just keep the parameter type as-is?

Reply:
@ardatan nodeRequest argument in getPort is coming from normalizeNodeRequest in const rawRequest = nodeRequest.raw || nodeRequest.req || nodeRequest, which is a union type `NodeRequest | IncomingMessage | Http2ServerRequest`